thorlcr/thorutil/thmem.cpp

```diff
@@ -245,11 +245,13 @@ class CSpillableStreamBase : public CSpillable
         StringBuffer tempName;
         VStringBuffer tempPrefix("streamspill_%d", activity.queryId());
         GetTempFilePath(tempName, tempPrefix.str());
-        spillFile.setown(createIFile(tempName.str()));
+        OwnedIFile iFile = createIFile(tempName.str());
+        spillFile.setown(new CFileOwner(iFile.getLink(), activity.queryTempFileSizeTracker()));
 
         VStringBuffer spillPrefixStr("SpillableStream(%u)", spillPriority);
-        rows.save(*spillFile, spillCompInfo, false, spillPrefixStr.str()); // saves committed rows
+        rows.save(*iFile, spillCompInfo, false, spillPrefixStr.str()); // saves committed rows
         rows.kill(); // no longer needed, readers will pull from spillFile. NB: ok to kill array as rows is never written to or expanded
+        spillFile->noteSize(iFile->size());
```

Comment:
When faster updates are revisited, may want to move this into save with a lower granularity (e.g. update ever second or every 1k records, within save).

Reply:
Jira created for reminded: https://hpccsystems.atlassian.net/browse/HPCC-32048